### PR TITLE
feat: add _safe flavor APIs for metadata and partition-count

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

**Comment by ohmycto** [comment]
Hi there, could you tell when this will be merged? We need it 🙏

File: src/brod_client.erl [hunk 1]
```diff
@@ -573,6 +603,7 @@ do_get_metadata(Topic, #state{ client_id     = ClientId
     {ok, #kpro_rsp{api = metadata, msg = Metadata}} ->
       TopicMetadataArray = kf(topics, Metadata),
       ok = update_partitions_count_cache(Ets, TopicMetadataArray),
+      ok = maybe_chache_unknown_topic_partition(Ets, Topic, TopicMetadataArray),
```
Comment (id) [inline src/brod_client.erl]:
```suggestion
      ok = maybe_cache_unknown_topic_partition(Ets, Topic, TopicMetadataArray),
```

File: src/brod_client.erl [hunk 2]
```diff
@@ -751,11 +782,23 @@ is_cooled_down(Ts, #state{config = Config}) ->
   Diff = timer:now_diff(Now, Ts) div 1000000,
   Diff >= Threshold.
 
+%% call this function after fetched metadata for all topics
+%% to cache the not-found status of a given topic
+maybe_chache_unknown_topic_partition(Ets, Topic, TopicMetadataArray) ->
```
Comment (id) [inline src/brod_client.erl]:
```suggestion
maybe_cache_unknown_topic_partition(Ets, Topic, TopicMetadataArray) ->
```

**Review by id — APPROVED** [review]
nitpicking

P.S. clients could do that already by querying `all`?

**Comment by zmstone** [comment]
> nitpicking

Thanks

> P.S. clients could do that already by querying `all`?

Yes, but no result-caching.

**Comment by zmstone** [comment]
tagged `3.16.5` (but not `3.16.6`, I forgot that `3.16.5` was never released).
also published to hex.

**Pull request merged**